web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateSearchApiController.java

```diff
@@ -182,6 +190,108 @@ public ResponseEntity<LinkedHashSet<org.oncokb.oncokb_transcript.client.Drug>> s
         return new ResponseEntity<>(getLimit(new LinkedHashSet<>(oncokbTranscriptService.findDrugs(query)), limit), HttpStatus.OK);
     }
 
+     private List<TypeaheadSearchResp> findMatchingCancerTypes(String query, Boolean exactMatch) {
+        // for now only finding exact matches
+         // got all evidences and level
+         System.out.println("This got to cancer type search ");
+         Set<Evidence> evidences = EvidenceUtils.getEvidenceByEvidenceTypesAndLevels(EvidenceTypeUtils.getTreatmentEvidenceTypes(), LevelUtils.getPublicLevels());
+         Map<String, CancerTypeMatch> result = new HashMap<>();
+
+         query = query.toLowerCase();
+         System.out.println("This is the query: " + query);
+
+         ArrayList<TumorType> matchedTumorTypes = new ArrayList<>();
+
+         if (exactMatch){
+             matchedTumorTypes.add(ApplicationContextSingleton.getTumorTypeBo().getByName(query));
+         }
+
+         if (matchedTumorTypes.isEmpty() && !exactMatch) {
+             for (Map.Entry<String, TumorType> currSubType : CacheUtils.getLowercaseSubtypeTumorTypeMap().entrySet()) {
+                 if(currSubType.getKey().startsWith(query)){
+                     matchedTumorTypes.add(currSubType.getValue());
+                 } else if (currSubType.getKey().contains(query)){
+                     matchedTumorTypes.add(currSubType.getValue());
+                 }
+             }
+
+             for (Map.Entry<String, TumorType> currMainType : CacheUtils.getMainTypeTumorTypeMap().entrySet()) {
+                 if(currMainType.getKey().toLowerCase().startsWith(query)){
+                     matchedTumorTypes.add(currMainType.getValue());
+                 } else if (currMainType.getKey().toLowerCase().contains(query)){
+                     matchedTumorTypes.add(currMainType.getValue());
+                 }
+             }
+         }
+
+        // cancer type not found, return an empty list
+         if (matchedTumorTypes.isEmpty()){
+             System.out.println("Nothing found");
+             return Collections.emptyList();
+         }
+
+         // definitely a matching cancer type, now searching evidence for relevant ones
+         for (TumorType currMatchedCancer : matchedTumorTypes){
```

Comment:
If no evidence is associated, we should still return the cancer type as a match. But just say in the frontend that no actionable gene is associated with this cancer type.